atomicapp/providers/kubernetes.py

```diff
@@ -38,227 +44,234 @@ class KubernetesProvider(Provider):
+        self.k8s_artifacts = {}
 
         logger.debug("Given config: %s", self.config)
         if self.config.get("namespace"):
             self.namespace = self.config.get("namespace")
 
         logger.info("Using namespace %s", self.namespace)
-        if self.container:
-            self.kubectl = self._find_kubectl(Utils.getRoot())
-            kube_conf_path = "/etc/kubernetes"
-            host_kube_conf_path = Utils.get_real_abspath(kube_conf_path)
-            if not os.path.exists(kube_conf_path) and os.path.exists(host_kube_conf_path):
-                if self.dryrun:
-                    logger.info("DRY-RUN: link %s from %s" % (kube_conf_path, host_kube_conf_path))
-                else:
-                    os.symlink(host_kube_conf_path, kube_conf_path)
-        else:
-            self.kubectl = self._find_kubectl()
+
+        self._process_artifacts()
 
         if not self.dryrun:
-            if not os.access(self.kubectl, os.X_OK):
-                raise ProviderFailedException("Command: " + self.kubectl + " not found")
 
-            # Check if Kubernetes config file is accessible, but only
-            # if one was provided by the user; config file is optional.
-            if self.config_file:
-                self.checkConfigFile()
+            '''
+            Config_file:
+                If a config_file has been provided, use the configuration
+                from the file and load the associated generated file.
+                If a config_file exists (--provider-config) use that.
 
-    def _find_kubectl(self, prefix=""):
-        """Determine the path to the kubectl program on the host.
-        1) Check the config for a provider_cli in the general section
-           remember to add /host prefix
-        2) Search /usr/bin:/usr/local/bin
+            Params:
+                If any provider specific parameters have been provided,
+                load the configuration through the answers.conf file
 
-        Use the first valid value found
+            Default configuration:
+                If no paramaters have been provider, as well as --provider-config,
+                use the default .kube/config location and load the associated configuration.
+            '''
+            if self.config_file:
+                self.api = Client(KubeConfig.from_file(self.config_file), "kubernetes")
+            elif self._check_required_params():
+                self.api = Client(self._from_required_params(), "kubernetes")
+            else:
+                config_location = os.path.join(Utils.getRoot(), Utils.getUserHome().strip('/'), '.kube/config')
+                self.api = Client(KubeConfig.from_file(config_location), "kubernetes")
+
+            # Check if the namespace that the app is being deployed to is available
+            self._check_namespaces()
+
+    def _check_required_params(self):
+        '''
+        This checks to see if any parameters associated to the Kubernetes provider are passed
```

Comment:
@dustymabe this checks to see if any CLI params or answers have been passed. does that make sense? so when you pass `--provider-auth --provider-api --provider-cert, etc.` it will create a generate ~/.kube/config using that information. Similar to how the configuration is generated with the `openshift.py` provider.